### PR TITLE
Add LuaCheck and GitHub workflow

### PR DESCRIPTION
**Pull request by Panquesito7**
Things added/changed:

- Add LuaCheck and GitHub workflow.
  - You can see the workflow run here: https://github.com/Panquesito7/cannons/actions/runs/5581586771/jobs/10199895442

### CODE REVIEW

**Comment by SwissalpS** [comment]
Not importand as it is only in some silly comments, but I'm curious:

Is github not able to show "ö" (o-umlaut) or were you using some other encoding than utf8?

Eichhörnchen

Seems to display without problems here in the comments.

File: .luacheckrc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+unused_args = false
+allow_defined_top = true
```
Comment (OgelGames) [inline .luacheckrc]: Not needed. (really it should never be used, all globals should be explicitly defined)
```suggestion
```

**Comment by OgelGames** [comment]
> Is github not able to show "ö" (o-umlaut) or were you using some other encoding than utf8?

Looks like an encoding change:
image
Those comments could probably be removed though, not really of much benefit.

**Comment by Panquesito7** [comment]
Whoops, used the wrong commit message, sorry. 😅

File: .luacheckrc (hunk 1 above)
Comment (S-S-X) [reply .luacheckrc]: Second this, allow_defined_top should never be enabled for mods and not really many legit use cases anywhere.
This test failing (unless it is just some missing known globals) can indicate serious issues with global namespace pollution and should fail (and even Minetest itself will complain for this).

**Comment by BuckarooBanzay** [comment]
LGTM, thanks :+1:

**Pull request merged**